placeSpecificContent/hq/chatterboxes/chatterbox/buttons/privacyButton_server.js

```diff
@@ -0,0 +1,404 @@
+        },
+
+        /* Move entities and if they are all in place, end the transition */
+        updateEntities: function( doorMovements, shadeTargetHeight) {
+            var doorCurrentLocalPosition = Entities.getEntityProperties(door, 'localPosition').localPosition;
+
+            _this.maybeMoveDoorOnXAxis(doorMovements, doorCurrentLocalPosition);
+            _this.maybeMoveDoorOnZAxis(doorMovements, doorCurrentLocalPosition);
+            _this.maybeChangeShadeHeight(shadeTargetHeight);
+
+            if (!doorMovements.xVelocity && !doorMovements.zVelocity && shadeHeightChange === 0) {
+                _this.playSound(DONE_SOUND, DONE_VOLUME, buttonPosition, false, false);
+                currentState = (currentState === "toClosed") ? "closed" : "open";
+                Entities.editEntity(button, {
+                    color: currentState === "closed" ? BUTTON_CLOSED_COLOR : BUTTON_OPEN_COLOR
+                });
+                if (updateInterval) {
+                    Script.clearInterval(updateInterval);
+                    updateInterval = null;
+                }
+            }
+        },
+
+        /* If the door is moving, check its position along the x axis to see if it is place. If it is, check if we need 
+        to move it along the z axis and begin that movement if needed. */
+        maybeMoveDoorOnXAxis: function(doorMovements, doorCurrentLocalPosition) {
+            // If door is moving on x axis
+            if (doorMovements.xVelocity && doorMovements.currentAxis === "x") {
+                // If door is in place, stop x movement,delete the x movement from the JSON and begin z movement if necessary
+                if (doorMovements.xVelocity > 0 && doorCurrentLocalPosition.x >= doorMovements.targetPosition.x || 
+                        doorMovements.xVelocity < 0 && doorCurrentLocalPosition.x <= doorMovements.targetPosition.x) {
+                    doorCurrentLocalPosition = {
+                        x: doorMovements.targetPosition.x,
+                        y: doorMovements.targetPosition.y,
+                        z: doorCurrentLocalPosition.z
+                    };
+                    delete doorMovements.xVelocity;
+                    if (doorMovements.zVelocity) {
+                        doorMovements.currentAxis = "z";
+                        Entities.editEntity(door, {
+                            localVelocity:  { x: 0, y: 0, z: doorMovements.zVelocity },
+                            localPosition: doorCurrentLocalPosition
+                        });
+                    } else {
+                        Entities.editEntity(door, {
+                            localVelocity: { x: 0, y: 0, z: 0 },
+                            localPosition: doorMovements.targetPosition
+                        });
+                    }
+                } 
+            }
+        },
+
+        /* If the door is moving, check its position along the z axis to see if it is place. If it is, check if we need 
```

Comment:
I'm confused about why this might need to move along the X axis. Can you please help me understand?

Reply:
same as above